cmake/TestSetup.cmake

```diff
@@ -69,16 +69,41 @@ function(setup_test_coverage dir arg)
   endif()
 endfunction()
 
+function(process_mpi_args nproc)
```

Comment:
This might be just a cmake misunderstanding on my part: are we using `nproc` in this function? Does it need to be passed in?

Reply:
It's used further down in line 88.